lib/gateways/gemfile.rb

```diff
@@ -3,8 +3,12 @@
 module Gateways
   class Gemfile
     def execute(application_name:)
-      open(gemfile_url(application_name)) do |gemfile_contents|
-        Domain::Gemfile.new(file_contents: gemfile_contents.read)
+      begin
+        open(gemfile_url(application_name)) do |gemfile_contents|
+          Domain::Gemfile.new(file_contents: gemfile_contents.read)
+        end
+      rescue(OpenURI::HTTPError) => _
```

Comment:
Feels kinda weird it fails silently, feels like the usecase could be the one rescuing from this exception, and here we just throw our own one like `GemfileNotFoundException` or something. Not sure what the usual approach is